conn.go

```diff
@@ -476,17 +495,58 @@ func (c *Conn) handleData(arg string) {
 	}
 
 	if c.server.LMTP {
-		// TODO: support per-recipient responses
 		for _, rcpt := range c.recipients {
-			c.WriteResponse(code, enhancedCode, "<"+rcpt+"> "+msg)
+			var status *SMTPError
+			rcptStatus := statuscollector.rcptStatus[rcpt]
+			select {
+			case <-rcptStatus.ctx.Done():
+				c.Server().ErrorLog.Printf("Context Error: %s - tempfailing", rcptStatus.ctx.Err())
+				tempfail := SMTPError{
+					Code:         420,
+					EnhancedCode: EnhancedCode{4, 4, 7},
+					Message:      "Error: timeout reached",
+				}
+				status = &tempfail
+			case status = <-rcptStatus.ch:
+			}
+			c.WriteResponse(status.Code, status.EnhancedCode, "<"+rcpt+"> "+status.Message)
 		}
+
 	} else {
 		c.WriteResponse(code, enhancedCode, msg)
 	}
 
 	c.reset()
 }
 
+type rcptStatus struct {
+	ctx context.Context
+	ch  chan *SMTPError
+}
+
+type lmtpStatusCollector struct {
+	rcptStatus map[string]*rcptStatus
+}
+
+func newlmtpStatusCollector(log Logger) *lmtpStatusCollector {
+	return &lmtpStatusCollector{
+		rcptStatus: make(map[string]*rcptStatus),
+	}
+}
+
+func (s *lmtpStatusCollector) SetStatus(rcpt string, status *SMTPError) {
+	rcpt = strings.ToLower(rcpt)
+	s.rcptStatus[rcpt].ch <- status
+}
+
+func (s *lmtpStatusCollector) StartDelivery(ctx context.Context, rcpt string) {
```

Comment:
Still not sure why calling this is necessary.

Reply:
its required to have a start point to create the channel and also assign the ctx, each rcpt can have a own ctx child with seperate timeout/deadline

Reply:
Can we leave the context for later? There are too many new things in this PR.

>create the channel

We can do that when creating the `lmtpStatusCollector`

Reply:
when creating lmtpStatusCollector/ (refactored to dataContext), we dont have a ctx per rcpt. Without a ctx we cannot implement a timeout, but this is highly needed

Reply:
1. Isn't one context for all recipients enough?
2. Why do we need to add this to go-smtp? Can't backends implement a timeout internally?
3. As said above, I'm not a fan of "big pull requests" adding a lot of new features.

Reply:
currently we dont have a context support in the library, There should be a way to get the ctx from the library user ( i think StartDelivery is a elegant way, you can use the same ctx as in a http library ), its not forced to use a ctx for each rcpt, but its possible

it isnt that big pull request, I rebased it into one commit, we just add the feature "multi rcpt status", of couse with ctx/channel support, if a user doesnt want to parallel delivery to his backend he just have to not use go() routines in his code. we collected the status just with channels thats the go way.